internal/bgp/native/native.go

```diff
@@ -49,7 +49,7 @@ type session struct {
 	closed         bool
 	conn           net.Conn
 	actualHoldTime time.Duration
-	defaultNextHop net.IP
+	nextHop        net.IP
```

Comment:
if you are removing this field I will remove it from the structure and the api below and just use addr.IP in the update msg, so we will have no reference to nexthop in the native stack

Reply:
We can't remove it as it's used in sendUpdate below.
See https://github.com/metallb/metallb/pull/1119/files#diff-cd8c067f48fafd2536808963cdc4b62bacf5309bdf396790de744e793a239362R145

Reply:
I was thinking of getting it internally when its needed instead of passing it around 
```
addr, ok := conn.LocalAddr().(*net.TCPAddr)
	if !ok {
		conn.Close()
		return fmt.Errorf("getting local addr for default nexthop to %q: %s", s.addr, err)
	}
	s.nextHop = addr.IP
```

but I am not really sure if this cleaner or what  u have is cleaner I am fine either way

Reply:
I understood, but there's another function, different from this, that uses the field and there we don't have addr.IP available.
See https://github.com/metallb/metallb/pull/1119/files#diff-cd8c067f48fafd2536808963cdc4b62bacf5309bdf396790de744e793a239362R126

Reply:
`func (s *session) sendUpdates() bool {`

Reply:
you can find the ip value using the code snippet I shared getIP() anyway just a thought I am fine with ur approach just wanted to bounce with u

Reply:
Now I get it, we could still do it because conn is stored in the session object. However, it doesn't cost too much and I think it is a bit more clear if we have the field named properly.